source/concepts.tex

```diff
@@ -0,0 +1,1249 @@
+
+  // \ref{concepts.lib.corelang.assignable}, Assignable:
+  template <class LHS, class RHS>
+  concept Assignable = @\seebelow@;
+
+  // \ref{concepts.lib.corelang.swappable}, Swappable:
+  template <class T>
+  concept Swappable = @\seebelow@;
+
+  template <class T, class U>
+  concept SwappableWith = @\seebelow@;
+
+  // \ref{concepts.lib.corelang.destructible}, Destructible:
+  template <class T>
+  concept Destructible = @\seebelow@;
+
+  // \ref{concepts.lib.corelang.constructible}, Constructible:
+  template <class T, class... Args>
+  concept Constructible = @\seebelow@;
+
+  // \ref{concepts.lib.corelang.defaultconstructible}, DefaultConstructible:
+  template <class T>
+  concept DefaultConstructible = @\seebelow@;
+
+  // \ref{concepts.lib.corelang.moveconstructible}, MoveConstructible:
+  template <class T>
+  concept MoveConstructible = @\seebelow@;
+
+  // \ref{concepts.lib.corelang.copyconstructible}, CopyConstructible:
+  template <class T>
+  concept CopyConstructible = @\seebelow@;
+
+  // \ref{concepts.lib.compare}, comparison concepts:
+  // \ref{concepts.lib.compare.boolean}, Boolean:
+  template <class B>
+  concept Boolean = @\seebelow@;
+
+  // \ref{concepts.lib.compare.equalitycomparable}, EqualityComparable:
+  template <class T>
+  concept EqualityComparable = @\seebelow@;
+
+  template <class T, class U>
+  concept EqualityComparableWith = @\seebelow@;
+
+  // \ref{concepts.lib.compare.stricttotallyordered}, StrictTotallyOrdered:
+  template <class T>
+  concept StrictTotallyOrdered = @\seebelow@;
+
+  template <class T, class U>
+  concept StrictTotallyOrderedWith = @\seebelow@;
+
+  // \ref{concepts.lib.object}, object concepts:
+  // \ref{concepts.lib.object.movable}, Movable:
+  template <class T>
+  concept Movable = @\seebelow@;
+
+  // \ref{concepts.lib.object.copyable}, Copyable:
+  template <class T>
+  concept Copyable = @\seebelow@;
+
+  // \ref{concepts.lib.object.semiregular}, Semiregular:
+  template <class T>
+  concept Semiregular = @\seebelow@;
+
+  // \ref{concepts.lib.object.regular}, Regular:
+  template <class T>
+  concept Regular = @\seebelow@;
+
+  // \ref{concepts.lib.callable}, callable concepts:
+  // \ref{concepts.lib.callable.invocable}, Invocable:
+  template <class F, class... Args>
+  concept Invocable = @\seebelow@;
+
+  // \ref{concepts.lib.callable.regularinvocable}, RegularInvocable:
+  template <class F, class... Args>
+  concept RegularInvocable = @\seebelow@;
+
+  // \ref{concepts.lib.callable.predicate}, Predicate:
+  template <class F, class... Args>
+  concept Predicate = @\seebelow@;
+
+  // \ref{concepts.lib.callable.relation}, Relation:
+  template <class R, class T, class U>
+  concept Relation = @\seebelow@;
+
+  // \ref{concepts.lib.callable.strictweakorder}, StrictWeakOrder:
+  template <class R, class T, class U>
+  concept StrictWeakOrder = @\seebelow@;
+}
+\end{codeblock}
+
+\rSec1[concepts.lib.corelang]{Core language concepts}
```

Comment:
I don't think we call this "core language" in the standard ever. We have a "language support library", so maybe these are just "language-related concepts"?